theano/gof/vm.py

```diff
@@ -92,10 +92,18 @@ def calculate_reallocate_info(order, fgraph, storage_map, compute_map_re, depend
                 viewed_by[origin].append(out)
             idx_o += 1
 
+        if hasattr(node.fgraph, 'shape_feature'):
+            same_shape = node.fgraph.shape_feature.same_shape
+        else:
+            def same_shape(x, y):
+                if x.ndim == 0 and y.ndim == 0:
+                    return True
+                else:
+                    return False
+
         for ins in node.inputs:
             assert not (ins in view_of and viewed_by[ins])
-            if (getattr(ins, 'ndim', None) == 0 and not storage_map[ins][0]
-                    and ins not in fgraph.outputs and ins.owner
+            if (storage_map[ins][0] is None and ins not in fgraph.outputs and ins.owner
```

Comment:
Can you do the fix for storage_map and None to the old code in the profiling of the memory?